plugins/woocommerce/includes/admin/helper/class-wc-helper-api.php

```diff
@@ -163,9 +166,9 @@ public static function put( $endpoint, $args = array() ) {
 	 *
 	 * @return string The absolute endpoint URL.
 	 */
-	public static function url( $endpoint ) {
+	public static function url( $endpoint, $source = '' ) {
 		$endpoint = ltrim( $endpoint, '/' );
-		$endpoint = sprintf( '%s/%s', self::$api_base, $endpoint );
+		$endpoint = sprintf( '%s/%s/%s', self::$api_base, $endpoint, $source );
```

Comment:
I wonder if it's worth trimming off any trailing `/`, in case `$source` is `''`?

```suggestion
		$endpoint = sprintf( '%s/%s/%s', self::$api_base, $endpoint, $source );
		$endpoint = rtrim( $endpoint, '/' );
```

Reply:
From my testing no request that hits the subscriptions endpoint on WooCommerce.com ends in a `/`, regardless whether or not it includes a query string, but no harm it being cautious, I think.